docker/README.md

```diff
@@ -63,9 +63,60 @@ Jenkins Kubernetes plugin slaves:
 
 -  [android-slave](./android-slave)
 
-## Build
+### Build
 
-To build slave execute docker build
+We are using s2i technology to for building mobile jenkins slaves. During the s2i build, a specific version of the Android SDK is installed and the user is asked to confirm all required licenses.
 
-    docker build . -t aerogear/jenkins-android-slave
+The android slave extends the openshift/jenkins-slave-base-centos7
+Any change in s2i dockerfile would require image build.
 
+    cd android-slave directory
+    docker build -t aerogear/jenkins-android-slave-s2i
+
+To include the android-sdk, download the required sdk, install and accept the license agreement
+The following script is an example of downloading the sdk installation
+
+    ```
+    wget --output-document=android-sdk.tgz --quiet https://dl.google.com/android/android-sdk_r24.4.1-linux.tgz
+    tar xzf android-sdk.tgz 
+    
+    android-sdk-linux/tools/android update sdk --all --no-ui --filter platform-tools,tools,build-tools-25.0.0,android-25,addon-google_apis_x86-google-21,extra-android-support,extra-google-google_play_services
+
+    ```
+
+    s2i build <directory-where-sdk-has-been-installed> aerogear/jenkins-android-slave-s2i aerogear/jenkins-android-slave:<version>
+
+For release builds, publish the image to the openshift internal registry - please refer to this link for more info
+https://docs.openshift.com/container-platform/3.3/dev_guide/managing_images.html
+
+### Android slave s2i image 
+
+The s2i build requirements (refer to the documentation here https://docs.openshift.com/enterprise/3.2/creating_images/s2i.html) must have both 'assemble and run' script in order to execute correctly.
+As this extends the openshift/jenkins-slave-base-centos7 image we needed to override the original entry point (https://github.com/openshift/jenkins/blob/master/slave-base/Dockerfile#L26), to execute the actual s2i build.
+The script (see below) will first check the params that are passed to it when executing the actual `s2i build` command, if the requirements are met it will follow the normal build flow.
+When called by the slave with parameters the original entry point script will be called.
+
+```
```

Comment:
-1 for duplication. Can we reference file here. Let's keep things simple and.. short.